gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,335 @@
+gpii.processes.findSolutionsByCommands = function (commandNames) {
+    if (!!commandNames) {
+        return fluid.accumulate(commandNames, function (aCommand, matches) {
+            var procInfos = gpii.processes.findProcessesByCommand(aCommand);
+            matches = matches.concat(procInfos);
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+// Return an list of process information structures corresponding to the
+// pids pass in.
+gpii.processes.findSolutionsByPids = function (pids) {
+    if (!!pids) {
+        return fluid.accumulate(pids, function (aPid, matches) {
+            var found = gpii.processes.findProcessByPid(aPid);
+            if (found !== null) {
+                matches.push(found);
+            }
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+// Return THE process info object that matches the given process id.
+// Note that it can return "null" meaning there is no such process.
+gpii.processes.findProcessByPid = function (pid, procArray) {
+    if (!procArray) {
+        procArray = gpii.processes["native"].getProcessList(pid);
+    }
+    return fluid.find(procArray, function (procInfo) {
+        if (procInfo.pid === pid) {
+            return procInfo;
+        }
+    }, null);
+};
+
+// Return an array of process information objects that match the given
+// command name string.  Note that it can return an empty array, meaning
+// there is no such process.
+gpii.processes.findProcessesByCommand = function (commandName, procArray) {
+    if (!procArray) {
+        procArray = gpii.processes["native"].getProcessList(commandName);
+    }
+    return fluid.accumulate(procArray, function (aProcInfo, matchingProcs) {
+        if (aProcInfo.command === commandName) {
+            matchingProcs.push(aProcInfo);
+        }
+        return matchingProcs;
+    }, []);
+};
+
+// Return the first process of an array of processes all with the same
+// command name string.  If there are no matching processes, return null.
+gpii.processes.findFirstProcessByCommand =
+function (commandName, procArray) {
+    var commands =
+        gpii.processes.findProcessesByCommand(commandName, procArray);
+    if (commands.length > 0) {
+        return commands[0];
+    }
+    else {
+        return null;
+    }
+};
+
+// Determine if the state of the given process has changed.  Record the
+// new process information in the monitor.  Return boolean.
+gpii.processes.hasStateChanged = function (monitor) {
+    if (!monitor || !monitor.procInfo) {
+        return false;    // nothing sought === nothing changed.
+    }
+    monitor.newProcInfo =
+    gpii.processes.updateProcInfo(monitor.procInfo);
+    return (monitor.procInfo.state !== monitor.newProcInfo.state);
+};
+
+// Utility function to conflate process state values into "Running"
+// (= true) vs. "Not Running" (= false).  Returns boolean.
+gpii.processes.isRunning = function (state) {
```

Comment:
A static JSON lookup would be more idiomatic than a switch - 
return gpii.processes.stateToRunning[state];